File: backend/app/api/schemas/copywriter.py
from __future__ import annotations

from pydantic import BaseModel, Field


class CopywriterRequest(BaseModel):
    """Entrada para gerar um e-mail. Todos os campos do seu prompt."""
    empresa: str = Field(..., description="Nome da empresa alvo")
    segmento: str | None = Field(default=None, description="Nicho/setor")
    nome_contato: str | None = Field(default=None, description="Nome do decisor")
    cargo: str | None = Field(default=None, description="Cargo do contato")
    canal: str = Field(
        "email",
        description="Canal de envio: email | whatsapp",
    )
    tipo: str = Field(
        "prospeccao",
        description="prospeccao | vaga | parceria | freelancer",
    )
    necessidade: str | None = Field(None, description="Dor identificada")
    servico: str | None = Field(
        None, description="O que está sendo ofertado")
    diferenciais: str | None = Field(
        None, description="Diferenciais da Reative")
    contexto_extra: str | None = Field(
        None,
        description="Texto livre — ex: descrição de vaga colada de um site de freelancer",
    )
    # Permite gerar a partir de um lead já existente no histórico
    lead_arquivo: str | None = Field(
        None, description="Nome do arquivo em data/sent/ para puxar contexto"
    )


class EmailGerado(BaseModel):
    assunto: str
    corpo: str
    cta: str
    tom: str


class CopywriterResponse(BaseModel):
    success: bool = True
    email: EmailGerado
    variantes: list[EmailGerado] = Field(
        default_factory=list,
        description="Versões alternativas (A/B) do mesmo e-mail",
    )

File: frontend/lib/types/copywriter.ts
// Copywriter / Outreach — geração de e-mails, follow-ups e histórico de envio.

export interface CopywriterRequest {
  empresa: string;
  segmento?: string | null;
  nome_contato?: string | null;
  cargo?: string | null;
  tipo: string;
  canal?: string;
  necessidade?: string | null;
  servico?: string | null;
  diferenciais?: string | null;
  contexto_extra?: string | null;
  lead_arquivo?: string | null;
}

export interface EmailGerado {
  assunto: string;
  corpo: string;
  cta: string;
  tom: string;
}

export interface CopywriterResponse {
  success: boolean;
  email: EmailGerado;
  variantes: EmailGerado[];
}

export interface GerarFollowupsRequest {
  dias?: number;
  max_followups?: number;
  limit?: number | null;
  pausa?: number;
}

export interface GerarFollowupsResponse {
  success: boolean;
  gerados: number;
  falhas: number;
}

export interface GerarRascunhosResponse {
  success: boolean;
  gerados: number;
  falhas: number;
  pulados: number;
}

export interface SyncResponse {
  success: boolean;
  enviados_confirmados: number;
  respostas_detectadas: number;
}

export type EmailStatus =
  | 'rascunho'
  | 'enviado'
  | 'respondido'
  | 'sem_resposta'
  | 'erro';

export interface EmailItem {
  id: string;
  destinatario: string;
  assunto: string;
  tom: string | null;
  status: EmailStatus;
  follow_up_num: number;
  draft_criado_em: string | null;
  enviado_em: string | null;
  primeira_resposta_em: string | null;
}

export interface EmailHistoryResponse {
  items: EmailItem[];
  total: number;
}
